modAionImpl/src/org/aion/zero/impl/sync/handler/BlockPropagationHandler.java

```diff
@@ -163,8 +164,8 @@ public PropStatus processIncomingBlock(
             }
             boolean stored = blockchain.storePendingStatusBlock(block);
             if (stored) {
-                this.syncStats.updatePeerStoredBlocks(_displayId, 1);
-                this.syncStats.updatePeerTotalBlocks(_displayId, 1);
+                this.syncStats.updatePeerBlocks(displayId, 1, BlockType.STORED);
+                this.syncStats.updatePeerBlocks(displayId, 1, BlockType.RECEIVED);
```

Comment:
This ends up counting the received blocks twice because they are already recorded by `ResStatusHandler.java`. Both received updates should be removed.